frontend/src/components/App/Notifications/index.tsx

```diff
@@ -225,6 +229,20 @@ export default function Notifications() {
     }
   }, [events, notifications]);
 
+  useEffect(() => {
+    const currentAreAllNotificationsInDeleteState =
+      notifications.filter(notification => !notification.deleted).length === 0;
+    setAreAllNotificationsInDeleteState(currentAreAllNotificationsInDeleteState);
+    setAreThereUnseenNotifications(
+      notifications.filter(notification => notification.seen !== true).length > 0
+    );
+    if (currentAreAllNotificationsInDeleteState) {
+      setFilteredNotifications([]);
+    } else {
+      setFilteredNotifications(notifications.slice(0, maxNotificationsInPopup));
+    }
+  }, [notifications]);
```

Comment:
Since we are not yet on React 18, where state setting like these are batched together automatically, I think we can make it slightly more performant if we use a `useMemo` and return all the values we are processing here:
`const [areThereUnseenNotifications, areAllNotificationsInDeleteState, filteredNotifications] = useMemo()...`

Reply:
changed to memo